contracts/ContributionAccountingToken.sol

```diff
@@ -0,0 +1,77 @@
+// SPDX-License-Identifier: MIT
+// Compatible with OpenZeppelin Contracts ^5.0.0
+pragma solidity ^0.8.20;
+
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/extensions/ERC20Permit.sol";
+import "@openzeppelin/contracts/access/AccessControl.sol";
+
+contract ContributionAccountingToken is ERC20, ERC20Permit, AccessControl {
+    bytes32 public constant MINTER_ROLE = keccak256("MINTER_ROLE");
+    
+    uint256 public maxSupply;
+    uint256 public thresholdSupply;
+    uint256 public maxExpansionRate;
+    bool public transferRestricted = true;
+    
+    uint256 public lastMintTimestamp;
+
+    constructor(
+        address defaultAdmin,
+        uint256 _maxSupply,
+        uint256 _thresholdSupply,
+        uint256 _maxExpansionRate
+    ) ERC20("Contribution Accounting Token", "CAT") ERC20Permit("Contribution Accounting Token") {
```

Comment:
Each token should have its own name and its own symbol, chosen by the user. Therefore, instead of "Contribution Accounting Token" and "CAT", we should take the name and the symbol as parameters in the constructor.